docs/source/en/training/adapt_a_model.mdx

```diff
@@ -0,0 +1,40 @@
+# Adapt a model to a new task
+
+Many diffusion systems share the same components, allowing you to adapt a pretrained model for one task to an entirely different task.
+
+This guide will show you how to adapt a pretrained text-to-image model for inpainting. You'll initialize a pretrained [`UNet2DConditionModel`] and modify it's architecture to enable training for inpainting.
```

Comment:
Are there other components that we can adapt? For example, if you want a text-to-image model to do upscaling, can you do something like the following (basically initialize the vae with these [arguments](https://huggingface.co/stabilityai/stable-diffusion-x4-upscaler/blob/main/vae/config.json)) and then finetune it for upscaling?

```py
from diffusers import AutoencoderKL
model_id = "stabilityai/stable-diffusion-2-1"
vae = AutoencoderKL.from_pretrained(
    model_id, subfolder="vae", scaling_factor=0.08333,
    ...
)

Reply:
Think for now it's mostly the unet mainly by doing text-to-image to inpainting like shown here. Think this example suffices for now - it's very nicely written!

Reply:
Think VAE is not really fine-tuned